wsmaster/codenvy-hosted-system-ram/src/main/java/com/codenvy/service/systemram/DockerBasedSystemRamInfoProvider.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.service.systemram;
+
+import com.google.inject.Inject;
+
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.SystemInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.eclipse.che.commons.lang.Size.parseSize;
+
+/**
+ * Implementation of {@link SystemRamInfoProvider} based on docker.
+ *
+ * @author Igor Vinokur
+ */
+public class DockerBasedSystemRamInfoProvider implements SystemRamInfoProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DockerBasedSystemRamInfoProvider.class);
+
+    private static final String SYSTEM_RAM_INFO_ERROR = "An error occurred while getting system RAM info.";
+
+    private final DockerConnector dockerConnector;
+
+    @Inject
+    public DockerBasedSystemRamInfoProvider(DockerConnector dockerConnector) {
+        this.dockerConnector = dockerConnector;
+    }
+
+    @Override
+    public SystemRamInfo getSystemRamInfo() throws ServerException {
+        long systemRamUsed = 0;
+        long systemRamTotal = 0;
+        SystemInfo systemInfo;
+        List<String> allNodesRamUsage = new ArrayList<>();
+
+        try {
+            systemInfo = dockerConnector.getSystemInfo();
+        } catch (IOException e) {
+            LOG.error("Failed to retrieve system information from docker.", e);
+            throw new ServerException(SYSTEM_RAM_INFO_ERROR, e);
+        }
+        String[][] systemValues = systemInfo.getDriverStatus() == null ? systemInfo.getSystemStatus() : systemInfo.getDriverStatus();
```

Comment:
Mb it would be better to save systemInfo.getDriverStatus() to variable to aviod double invocation of it.